### PR TITLE
Restructure Metric Logging and Add support for W&B

### PR DESCRIPTION
**Pull request by manangoel99**
This PR redesigns the logging system in PaddleOCR without changing any of the console logging. The proposed change is to have a `BaseLogger` class from which new loggers can inherit. This includes the VisualDL logger, the new Weights & Biases Logger and any new custom loggers that might be added later. 
Using the W&B Logger would be really easy for users. The only requirement is to set `use_wandb: True` in the `Global` config.yml file. To add any other `wandb` related arguments like project name, entity name and so on, the users can add the `wandb` key.
For example:
```
Global:
    use_wandb: True

wandb:
    project: PaddleOCR
    entity: MyTeam
```
This leads to all the training metrics being logged to W&B. The dashboard:
Screenshot 2022-03-30 at 8 57 00 PM
Screenshot 2022-03-30 at 8 57 13 PM

The models at every model saving step and evaluation step are also logged to the W&B run with the appropriate tags and metadata.
Screenshot 2022-03-30 at 8 59 08 PM

An example dashboard is available [here](https://wandb.ai/manan-goel/PaddleOCR/runs/27e7nb40)

### CODE REVIEW

File: ppocr/utils/loggers/vdl_logger.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        updated_metrics = {prefix + "/" + k: v for k, v in metrics.items()}
+        
+        for k, v in updated_metrics.items():
+            self.vdl_writer.add_scalar(k, v, step)
+    
+    def close(self):
+        self.vdl_writer.close()
+
+
+
```
Comment (GreatV) [inline ppocr/utils/loggers/vdl_logger.py:21]: No newline at end of file.

**Comment by manangoel99** [comment]
Hi @GreatV ! Thank you for the quick response and review. I have added some documentation as well in english. It would be awesome if you could help me with the chinese version as well.
Do let me know if you think any other changes need to be made to the PR.

**Comment by GreatV** [comment]
> Hi @GreatV ! Thank you for the quick response and review. I have added some documentation as well in english. It would be awesome if you could help me with the chinese version as well. Do let me know if you think any other changes need to be made to the PR.

Hi @manangoel99 ! I'd be happy to contribute Chinese documentation for this part. And I have no further questions about this PR at the moment, please wait for comments from other reviewers and developers.

**Comment by manangoel99** [comment]
@LDOUBLEV @WenmuZhou @GreatV Please have a look at the PR whenever you get a chance. I'd be happy to make any changes if required.

**Comment by WenmuZhou** [comment]
Thanks pr, there are a few things that need to be changed
1.  the corresponding Chinese documents need to be modified
2. Added a tutorial on how to view logs in wandb
3. cherrypick to the dygraph branch

**Comment by manangoel99** [comment]
@WenmuZhou would it be alright if I make a new PR based out of dygraph because cherry picking is getting very messy because of merge conflicts between release/2.4 and dygraph?

**Comment by WenmuZhou** [comment]
Let us discuss here https://github.com/paddlepaddle/paddleocr/pull/5957

**Comment by manangoel99** [comment]
Closing in interest of #5957

**Comment by yiranr** [comment]
Hi Manan Goel, I'm Erin from PaddlePaddle, Baidu. Can I ask for your email address so that we could talk about cooperation in details?

**Comment by manangoel99** [comment]
@yiranr Sure! You can get in touch with me at manan.goel@wandb.com

**Comment by yiranr** [comment]
> @yiranr Sure! You can get in touch with me at [manan.goel@wandb.com](mailto:manan.goel@wandb.com)

I sent an email to you. Hope to hear from you soon~

**Pull request closed without merging**